Shared/Samples/Measure distance in scene/MeasureDistanceInSceneView.swift

```diff
@@ -0,0 +1,183 @@
+                                from: measurements.verticalDistance
+                            )
+                        }
+                    }
+                    .overlay(alignment: .top) {
+                        // Instruction text.
+                        Text(instructionText)
+                            .frame(maxWidth: .infinity, alignment: .center)
+                            .padding(8)
+                            .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+                    }
+            }
+            
+            // Distance texts.
+            Text("Direct: \(directDistanceText)")
+            Text("Horizontal: \(horizontalDistanceText)")
+            Text("Vertical: \(verticalDistanceText)")
+            
+            // Unit system picker.
+            Picker("", selection: $unitSystemSelection) {
+                Text("Imperial").tag(UnitSystem.imperial)
+                Text("Metric").tag(UnitSystem.metric)
+            }
+            .pickerStyle(.segmented)
+            .padding()
+            .onChange(of: unitSystemSelection) { _ in
+                model.locationDistanceMeasurement.unitSystem = unitSystemSelection
+            }
+        }
+    }
+}
+
+private extension MeasureDistanceInSceneView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A scene with an imagery basemap.
+        let scene = {
+            let scene = ArcGIS.Scene(basemapStyle: .arcGISTopographic)
+            
+            // Add elevation source to the base surface of the scene with the service URL.
+            let elevationSource = ArcGISTiledElevationSource(url: .worldElevationService)
+            scene.baseSurface.addElevationSource(elevationSource)
+            
+            // Create the building layer and add it to the scene.
+            let buildingsLayer = ArcGISSceneLayer(url: .brestBuildingsService)
+            scene.addOperationalLayer(buildingsLayer)
+            
+            return scene
+        }()
+        
+        /// An analysis overlay for location distance measurement.
+        let analysisOverlay = AnalysisOverlay()
+        /// The location distance measurement.
+        let locationDistanceMeasurement = LocationDistanceMeasurement(
+            startLocation: Point(x: -4.494677, y: 48.384472, z: 24.772694, spatialReference: .wgs84),
+            endLocation: Point(x: -4.495646, y: 48.384377, z: 58.501115, spatialReference: .wgs84)
+        )
+        
+        /// A measurement formatter for converting the distances to strings.
+        let measurementFormatter: MeasurementFormatter = {
+            let measurementFormatter = MeasurementFormatter()
+            measurementFormatter.unitOptions = .providedUnit
+            measurementFormatter.numberFormatter.minimumFractionDigits = 2
+            measurementFormatter.numberFormatter.maximumFractionDigits = 2
+            return measurementFormatter
+        }()
```

Comment:
Formatter objects [should not be used](https://developer.apple.com/documentation/foundation/data_formatting) in Swift:

> When working in Swift, use `formatted` methods directly on the types you want to format, optionally using [FormatStyle](https://developer.apple.com/documentation/foundation/formatstyle) and its subtypes to customize formatter output. This approach supports dates, integers, floating-point numbers, measurements, sequences, and person name components. Foundation caches identically-configured formatter instances internally, allowing you to focus on your app’s formatting needs.

Also, localized strings/values should not be stored. They should be computed on-demand when needed since (parts of) the locale can change while the app is running, affecting how the information is displayed. In this case, the measurements themselves should be stored and then formatted on demand like so:

```swift
Text("Direct: \(directDistance, format: ...)")
...
```

As for the format, one way you can reuse the same format for each measurement is by extending format style:

```swift
extension FormatStyle where Self == Measurement<UnitLength>.FormatStyle {
    static var distance: Self {
        .measurement(width: .abbreviated, usage: .asProvided, numberFormatStyle: .number.precision(.fractionLength(2)))
    }
}
```

And then passing `.distance` as the format: `format: .distance`.

Reply:
@philium I updated the sample to use the format style, but can't get it work by storing the measurements. In using the following: 
```
for await measurements in model.locationDistanceMeasurement.measurements {
    directDistance = measurements.directDistance
    horizontalDistance = measurements.horizontalDistance
    verticalDistance = measurements.verticalDistance
}
```
The measurements updates come through the async stream just fine but in set them to the corresponding State variable, they wouldn't always update with the correct units when the unit system is toggled. Also, Changing the location distance measurement can cause some of distances to update when toggling the unit system but not all them. The aysnc stream and the API seem to be working fine but using `Measurement` State variables seems to be super buggy for some reason. Both Ting and looked at it and were at a loss on how to fix it so I just reverted back to using the localized strings. Let me know if there is better solution.

Reply:
That is surprising. I can't think of a better option at the moment, so what you have will have to do for now. Thank you for making those changes.